### PR TITLE
Fix list invites filter, validator and auth

### PR DESCRIPTION
**Pull request by edouard-sn**
fix: validator list invites

#### Description

Quick fix for invites filter and validator
Auth check for invites

Filter management was based on accounts `List` which turned out to be wrong

NOTE: If a user wants to list every one of his invite, he HAS to put his ID as recipient/sender ID, hope that's ok, it can always be changed

#### Changelog

- [ENHANCEMENT] Auth check for invites
- [ENHANCEMENT] Basic unit test for list invite
- [BUGFIX] List validator
- [BUGFIX] List filter

### CODE REVIEW

File: auth/test_service.go [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+	token := &Token{}
+	json.Unmarshal([]byte(tokenString), token)
+
+	return token, nil
+}
+
+// NOTE: Not sure but works for conversations tests
+func (srv *TestService) SignToken(info *Token) (string, error) {
+	return "", nil
+}
```
Comment (rojas-diego) [inline auth/test_service.go:53]: Yeah it's alright, I'll fix the interfaces next sprint!

File: models/mongo/accounts.go [hunk 2]
```diff
@@ -94,7 +94,7 @@ func (srv *accountsRepository) Delete(ctx context.Context, filter *models.OneAcc
 func (srv *accountsRepository) Update(ctx context.Context, filter *models.OneAccountFilter, account *models.AccountPayload) (*models.Account, error) {
 	var accountUpdated models.Account
 
-	field := bson.D{{"$set", bson.D{{"name", account.Name}}}}
+	field := bson.D{{Key: "$set", Value: bson.D{{"name", account.Name}}}}
```
Comment (rojas-diego) [inline models/mongo/accounts.go:97]: Haha we have so many warnings like that

File: validators/invites.go [hunk 3]
```diff
@@ -27,11 +27,7 @@ func ValidateDenyInvite(in *accountsv1.DenyInviteRequest) error {
 }
 
 func ValidateListInvites(in *accountsv1.ListInvitesRequest) error {
-	return validation.ValidateStruct(in,
-		validation.Field(&in.SenderAccountId, validation.Required, is.UUID),
-		validation.Field(&in.RecipientAccountId, validation.Required, is.UUID),
-		validation.Field(&in.GroupId, validation.Required, is.UUID),
-	)
+	return nil
```
Comment (rojas-diego) [inline validators/invites.go:30]: We still need to check that it is a valid UUID!

File: invites.go [hunk 4]
```diff
@@ -106,11 +106,38 @@ func (srv *invitesAPI) ListInvites(ctx context.Context, in *accountsv1.ListInvit
...
+		}
+
+		// If you are not trying to fetch invites from a group, you should only fetch those where you are sender or recipient
+	} else if in.RecipientAccountId != token.UserID.String() && in.SenderAccountId != token.UserID.String() {
+		return nil, status.Error(codes.Unauthenticated, "you have no rights to fetch other users invites")
+	}
+
 	if in.Limit == 0 {
 		in.Limit = 20
 	}
```
Comment (rojas-diego) [inline invites.go:143]: Line 127/154 you say 

> // NOTE: As every account are apparently accessible to list and get from every account as long as you are authentificated, I did the same with invites

Is that still d'actualité?

Comment (rojas-diego) [reply invites.go:143]: Same thing, maybe it's not a problem, but a bit further down in the endpoint you do
```go
	invites, err := srv.inviteRepo.List(ctx, &models.ManyInvitesFilter{
		SenderAccountID:    &in.SenderAccountId, // <--
		RecipientAccountID: &in.RecipientAccountId, // <--
		GroupID:            &in.GroupId, // <--
	}, &models.Pagination{Offset: int64(in.Offset), Limit: int64(in.Limit)})
	if err != nil {
		return nil, statusFromModelError(err)
	}
```
Shouldn't you be passing nil if the user has put no filter on sender, recipient or group? Does it not mess with the `models` layer? Just curious

File: invites.go [hunk 5]
```diff
@@ -106,11 +106,38 @@ func (srv *invitesAPI) ListInvites(ctx context.Context, in *accountsv1.ListInvit
...
+			// If you are not in the current group, you should ask only for invites where you are the recipient
+			if in.RecipientAccountId != token.UserID.String() {
+				return nil, status.Error(codes.Unauthenticated, "you can only fetch your invites for this group")
+			}
+		}
+
+		// If you are not trying to fetch invites from a group, you should only fetch those where you are sender or recipient
+	} else if in.RecipientAccountId != token.UserID.String() && in.SenderAccountId != token.UserID.String() {
+		return nil, status.Error(codes.Unauthenticated, "you have no rights to fetch other users invites")
+	}
```
Comment (rojas-diego) [inline invites.go:139]: Comments are super helpful!

Comment (rojas-diego) [reply invites.go:139]: I'm assuming the logic works

**Review by rojas-diego — DISMISSED** [review]
Looks good. I'm not sure if it's going to work in prod first try. Let's see!

File: invites.go (hunk 5 above)
Comment (edouard-sn) [reply invites.go:139]: Yeah, branch logic was kinda too much imo so i prefered to explained, thx :)

File: invites.go (hunk 4 above)
Comment (edouard-sn) [reply invites.go:143]: For first comment it is not anymore !

**Review by rojas-diego — APPROVED** [review]

**Pull request merged**